src/backend/metal/src/soft.rs

```diff
@@ -116,6 +116,13 @@ pub enum RenderCommand<R: Resources> {
         buffer: BufferPtr,
         offset: hal::buffer::Offset,
     },
+    InsertDebugMarker {
```

Comment:
We don't want to force owning of `String` here. This would mean an unconditional heap allocation.
Instead, let's extend `trait Resources` to work with strings, i.e:
```rust
trait Resources {
  type Data;
  type Marker;
  ...
}
impl Resources for Own {
    type Data = Vec<u32>;
    type Marker = String;
    ...
}
#[derive(Debug)]
pub struct Ref;
impl<'a> Resources for &'a Ref {
    type Data = &'a [u32];
    type Marker = &'a str;
    ...
}
```

This would allow us to pass in `&str`, and let it be converted to `String` only when necessary

Reply:
Yeah, the allocation was also making me uncomfortable but I didn't know any better :-).